ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,270 @@
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.ToolBox.PsychSignalDataConverter
+{
+    public class PsychSignalDataConverter
+    {
+        private readonly Dictionary<string, TickerData> _fileHandles;
+        private readonly HashSet<string> _knownTickers;
+        private readonly DirectoryInfo _rawSourceDirectory;
+        private readonly DirectoryInfo _destinationDirectory;
+
+        /// <summary>
+        /// Converts psychsignal raw data into a format usable by Lean
+        /// </summary>
+        /// <param name="sourceDirectory">Directory to source our raw data from</param>
+        /// <param name="destinationDirectory">Directory to write formatted data to</param>
+        /// <param name="knownTickerFolder">Directory where we source the ticker list</param>
+        public PsychSignalDataConverter(string sourceDirectory, string destinationDirectory, string knownTickerFolder)
+        {
+            _rawSourceDirectory = new DirectoryInfo(sourceDirectory);
+            _destinationDirectory = new DirectoryInfo(destinationDirectory);
+
+            _fileHandles = new Dictionary<string, TickerData>();
+            _knownTickers = Directory.GetFiles(knownTickerFolder, "*.zip").Select(Path.GetFileNameWithoutExtension).ToHashSet();
+
+            _destinationDirectory.Create();
+        }
+        
+        /// <summary>
+        /// Converts a specific file to Lean alternative data format. Note that you must flush
+        /// after you're done converting a file to ensure that all data gets written to disk.
+        /// You can do that by calling <see cref="FlushAll"/> once you've finished processing
+        /// 
+        /// Note: Assumes that it will be fed files in ascending order by date
+        /// </summary>
+        /// <param name="sourceFilePath">File to process and convert</param>
+        public void Convert(FileInfo sourceFilePath)
+        {
+            var file = File.ReadLines(sourceFilePath.FullName);
+
+            foreach (var line in file)
+            {
+                var csv = line.Split(',');
+                var ticker = csv[1].ToLower();
+                DateTime timestamp;
+
+                if (csv[0] == "SOURCE" || !_knownTickers.Contains(ticker) || !DateTime.TryParseExact(csv[2], @"yyyy-MM-dd\THH:mm:ss\Z", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out timestamp))
```

Comment:
nit - this may be a little more readable if each condition is on it's own line:
``` csharp
if (csv[0] == "SOURCE"
  || !_knownTickers.Contains(ticker)
  || !DateTime.TryParseExact(csv[2], @"yyyy-MM-dd\THH:mm:ss\Z", CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal, out timestamp))
```